src/pages/docs/git.md

```diff
@@ -0,0 +1,285 @@
+
+- Code Quality: this checks if your code follows your style guidelines. If this task fails, you can fix it by running `npm run format` on NextJS projects and `mix format` on Elixir projects. `npm run lint` and `mix lint` might also be needed
+
+- Build and Test: this only applies to backend projects, and it checks to make sure all the tests in the codebase run successfully (and that the project compiles, of course). You can simulate this task by running `mix test` on your local machine. If any tests fail, you will need to fix them before merging
+
+- Deploy Previews: deploys your branch on a temporary URL for testing purposes. If all other tasks pass, this one should not be a problem and, if it is, it probably is something DevOps related, so contact the person in charge of that.
+
+
+## Rebases, Merges and Conflicts
+
+If may happen that, in the time you opened your PR and someone approved it, another PR was merged and it causes conflicts with your code. This would be indicated in Github like the following image.
+
+To fix this, you can do two things: rebase your branch so it is up to date with `main` - we recommend doing this 90% of the time - or merging your branch with `main` - which we only recommend if the rebase process would be very difficult.
+
+### Rebase
+
+To rebase your branch start by updating your `main` branch by running:
+
+```
+git switch main
+git pull
+```
+
+Now switch back to your branch and start the rebase:
+
+```
+git switch <YOUR-BRANCH>
+git rebase main
+```
+
+Now go to your favourite text editor, and fix the conflicts yourself. Once you are done, add all the files and continue with the rebase.
+
+```
+git add <FILES CHANGED>
+git rebase --continue
+```
+
+You might have to repeat this a few times before the rebase succeeds. When it does, it is time to push your branch to the remote. However, a rebase is a rewriting of history, meaning you will have to force push by running
+
+```
+git push -f
+```
+
+### Merge
+
+To merge your branch start by updating your `main` branch by running:
+
+```
+git switch main
+git pull
+```
+
+Now switch back to your branch and start the merge:
+
+```
+git switch <YOUR-BRANCH>
+git merge main
+```
+
+Now go to your favourite text editor, and fix the conflicts yourself. Once you are done, add all the files and continue with the merge.
+
+```
+git add <FILES CHANGED>
+git merge --continue
+```
+
+Now it is time to push your branch to the remote.
+
+```
+git push
+```
+
+# Troubleshooting
+
+## I have commits that shouldn't be on my branch and want to delete them. How? (Interactive rebases)
+
+This can happen when you accidentally base your branch of another branch other than `main`, or if you have an erroneous commit in the middle of your history you want to delete. To do that, you simply update `main` and do an interactive rebase:
```

Comment:
```suggestion
This can happen when you accidentally base your branch off of another branch other than `main`, or if you have an erroneous commit in the middle of your history you want to delete. To do that, you simply update `main` and do an interactive rebase:
```